### ISSUE TITLE
session.getComponent doesn't work with 'default' keyword arg

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10667]
|<img alt="justinj's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| justinj reported|
|-|-|
|Trac ID|trac#992|
|Type|defect|
|Created|2005-04-19 01:38:14Z|

<details><summary>Searchable metadata</summary>

```
trac-id__992 992
type__defect defect
reporter__justinj justinj
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1113874694000000 1113874694000000
changetime__1166056932000000 1166056932000000
version__ 
owner__ 
cc__itamarst cc__justinj cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="justinj's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|justinj commented|
|-|-|

```
#!html
<pre>
After upgrading to Twisted 2.0 on Windows 2000, and changing code to use 
zope.interface, lines like following never pick up passed in default value.

thing = session.getComponent(IThing, default="blah")

Instead, thing is set to None.  Removing default= so that the default is just 
passed in as the 2nd arg fixes the problem.  Heard on irc that default keyword 
arg shouldn't be used anymore because zope doesn't support it.  I didn't see 
any doc on this anywhere, so I'm assuming it's a bug.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="jyknight's avatar" src="https://avatars.githubusercontent.com/u/873903?s=50" width="50" height="50">](https://github.com/jyknight)|@jyknight commented|
|-|-|

```
#!html
<pre>
Agree, looks like a bug in Componentized.getComponent. => itamarst.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Fixed in getcomponent-992.  Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @dreid|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)|@dreid set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)<a name="note_6"></a>|@dreid set owner to @exarkun<br>@dreid set status to `new`|
|-|-|

There isn't anything bad to note about this branch.  The change is simple and functional and the tests pass.  Ok to merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [18897]) Merge getcomponent-992

Author: exarkun
Reviewer: dreid
Fixes [#10667](https://github.com/twisted/twisted/issues/10667)

Add tests for twisted.python.components.Componentized.getComponent's
default argument and fix a bug which caused it to always be ignored.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
